### DISCUSSION TITLE
Prometheus per-object or not per-object

### DISCUSSION THREAD

**Discussion by miksir** [rabbitmq/rabbitmq-server#3061] [Questions] [answered]
Before all this happened it was standalone rabbitmq and rabbit-exporter.
But cluster coming and we decided to start to use rabbitmq metrics endpoint aka prometheus plugin.
And now I got dilemma. We need metrics per queue. Its really helps to find growing up queue fast, its helps to find queues without consumers. So aggregate mode not for us - there is no queue metrics. Per object mode.... well, lets skip the point of the amount of scraped metrics, it possible to fix with prometheus relabel.  But scrape time for us about 25-30 seconds, it's really hard to use especially if we want to scrape each 30 sec.
So, what to do... 
Probably switch to aggregate mode back and add default prometheus rabbitmq exporter, but I cant understand how good this exporter for cluster.
Any suggestions?

**Comment by daichi703n** [comment]
Hi, @miksir 
I have the same issue. Did you solved this? Or did you revert to the old exporter?

**Reply by miksir** [reply]
Using both, prometheus plugin without per object data and separate rabbit exporter for per queue metrics

**Reply by daichi703n** [reply]
Thank you. I would consider using both as well as you.

**Reply by mkuratczyk** [reply]
What RabbitMQ version are you using? I believe there are significant improvements made around November last year. Also, since 3.9.8, there is a configurable endpoint which allows to select only some of the per-object metrics: https://github.com/rabbitmq/rabbitmq-server/pull/3520

**Reply by daichi703n** [reply]
rabbitmq-server-3.9.11-1.el7.noarch.

> https://github.com/rabbitmq/cluster-operator/pull/865
> - QueueHasNoConsumers
> - QueueIsGrowing

These alerts are really what I want to do!

How can I enable detailed metrics? (not yet read docs...)

**Reply by mkuratczyk** [reply]
With that version, you should just have them (`/metrics/detailed` endpoint) and you already found the alerts you need that use that endpoint

**Reply by daichi703n** [reply, marked as answer]
Thanks. I successfully got each detailed metrics with query (`/metrics/detailed?family=xxx`) described here: https://github.com/rabbitmq/rabbitmq-server/tree/master/deps/rabbitmq_prometheus#selective-querying-of-per-object-metrics .

**Comment by michaelklishin** [comment]
It's not OK to hijack an existing discussion to ask new questions.

Perhaps start a new discussion instead, and if there are reasons to link to this one, do so.
